### PR TITLE
Add delegation support for RA calls

### PR DESCRIPTION
**Pull request by darshanhs90**
Linked to the following swagger specs pr
https://github.com/Azure/azure-rest-api-specs/pull/2031 
https://github.com/Azure/azure-rest-api-specs/pull/2043 

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by darshanhs90** [comment]
@shahabhijeet API specs PR has been merged

File: src/SDKs/Authorization/Authorization.Tests/Authorization.Tests.csproj [hunk 1]
```diff
@@ -10,9 +10,9 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Management.Authorization" Version="2.5.0-preview" />
+    <!--<PackageReference Include="Microsoft.Azure.Management.Authorization" Version="2.5.0-preview" />-->
```
Comment (shahabhijeet) [inline src/SDKs/Authorization/Authorization.Tests/Authorization.Tests.csproj]: @darshanhs90 remove commented code.
Also start using latest version of ResourceManager to record all your tests. The latest version is 1.6.0 for ResourceManager nuget package.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
@darshanhs90 regenerate your SDK from the merged spec.
When you do that, you will get an additional .txt file as part of your PR.

**Comment by darshanhs90** [comment]
@shahabhijeet Made all the changes as per the comments and rerecorded the tests

File: src/SDKs/Authorization/Authorization.Tests/Authorization.Tests.csproj (hunk 1 above)
Comment (darshanhs90) [reply src/SDKs/Authorization/Authorization.Tests/Authorization.Tests.csproj]: Resolved it

---
In reply to: [153349537](https://github.com/Azure/azure-sdk-for-net/pull/3872#discussion_r153349537) [](ancestors = 153349537)

**Comment by darshanhs90** [comment]
@shahabhijeet Fixed merge conflict issue and updated the version

File: src/SDKs/Authorization/Management.Authorization/Microsoft.Azure.Management.Authorization.csproj [hunk 2]
```diff
@@ -6,7 +6,7 @@
 	<PropertyGroup>
 		<Description>Provides capabilities to query Microsoft Azure Management Authorization.</Description>
 		<AssemblyTitle>Authorization Library for Microsoft Azure Management Authorization</AssemblyTitle>
-		<Version>2.5.0-preview</Version>
+		<Version>2.6.0-preview</Version>
```
Comment (shahabhijeet) [inline src/SDKs/Authorization/Management.Authorization/Microsoft.Azure.Management.Authorization.csproj]: @darshanhs90 please add packageRelease notes, it's left blank.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Authorization/Management.Authorization/Microsoft.Azure.Management.Authorization.csproj (hunk 2 above)
Comment (darshanhs90) [reply src/SDKs/Authorization/Management.Authorization/Microsoft.Azure.Management.Authorization.csproj]: @shahabhijeet Added package release notes back

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**